source/SongCore/HarmonyPatches/CosmeticCharacteristicsPatches.cs

```diff
@@ -0,0 +1,154 @@
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using HarmonyLib;
+using HMUI;
+using IPA.Utilities;
+using SongCore.Data;
+using SongCore.Utilities;
+using UnityEngine;
+using static IPA.Logging.Logger;
+
+namespace SongCore.HarmonyPatches
+{
+    internal class CosmeticCharacteristicsPatches
+    {
+
+
+        [HarmonyPatch(typeof(BeatLineManager))]
+        [HarmonyPatch(nameof(BeatLineManager.HandleNoteWasSpawned))]
+        internal class BeatLineManager_HandleNoteWasSpawned
+        {
+            private static bool Prefix()
+            {
+                if (Plugin.Configuration.DisableRotationSpawnLinesOverride)
+                    return true;
+
+                var beatmap = BS_Utils.Plugin.LevelData.GameplayCoreSceneSetupData.difficultyBeatmap;
+                if (beatmap == null)
+                    return true;
+                var beatmapData = Collections.RetrieveDifficultyData(beatmap);
+                if (beatmapData == null)
+                    return true;
+                if (beatmapData._showRotationNoteSpawnLines == null)
+                    return true;
+                return beatmapData._showRotationNoteSpawnLines.Value;
+            }
+        }
+
+
+        [HarmonyPatch(typeof(GameplayCoreInstaller))]
+        [HarmonyPatch(nameof(GameplayCoreInstaller.InstallBindings))]
+        internal class GameplayCoreInstaller_InstallBindingsPatch
+        {
+            private static ExtraSongData.DifficultyData? diffData = null;
+            private static int numberOfColors = -1;
+            private static GameplayCoreSceneSetupData sceneSetupData = null;
+            private static void Prefix(GameplayCoreInstaller __instance)
+            {
+                if (Plugin.Configuration.DisableOneSaberOverride)
+                    return;
+
+                sceneSetupData = __instance.GetField<GameplayCoreSceneSetupData, GameplayCoreInstaller>("_sceneSetupData");
```

Comment:
That could be replaced by direct access instead of using reflection, since `Main.dll` is publicized. It's probably the same for all of your `SetField` and `GetField` usages, but you can also publicize the assemblies you need in case there's one missing.